compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -109,6 +110,35 @@
     private Names names;
     private int finiteTypeCount = 0;
 
+    private boolean checkFillerValue(BUnionType type) {
```

Comment:
We seem to have methods with different names`checkFillerValue`, `hasFillerValue`, etc. across the class for similar checks. Shall we revisit these and group them together?

Reply:
hasFillerValue is the public method which calls checkfillerValue helper methods sometime recursively. I will try to refactor.